### PR TITLE
fix(frame): fixed an optimized the UTF8 validation for MQTT packets

### PR DESCRIPTION
**Pull request by lafirest**
Fixes EMQX-12559

Release version: v/e5.8.1

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx/src/emqx_frame.erl [hunk 1]
```diff
@@ -1261,40 +1261,10 @@ validate_utf8(Bin) ->
 %% i.e. contains invalid UTF-8 char or control char
```
Comment (zmstone) [inline apps/emqx/src/emqx_frame.erl:1261]:
```suggestion
%% i.e. does the string contains control characters?
%% Note: this is under the assumption that the string is already validated by `unicode:characters_to_binary/1`
%% hence there is no need to validate utf8 byte sequence integrity 
```

File: changes/ce/fix-13842.en.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Fixed and optimized UTF8 validation of MQTT packets in `strict` mode.
```
Comment (zmstone) [inline changes/ce/fix-13842.en.md:1]:
```suggestion
Fixed and UTF-8 string validation exception.
```

**Review by zmstone — APPROVED** [review]

**Review by HJianBo — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-4269)

File: changes/ce/fix-13842.en.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Fixed and UTF-8 string validation exception.
```
Comment (Le9Wang) [inline changes/ce/fix-13842.en.md:1]: typo: and -> a

Comment (zmstone) [reply changes/ce/fix-13842.en.md:1]: pushed a fix directly without sending a PR to release-58

**Pull request merged**